### PR TITLE
Fixed conversion warnings on framework tests

### PR DESCRIPTION
**Pull request by JohanBertrand**
| | |
|:---|:---|
|**_Related Issue(s)_**|  |
|**_Has Unit Tests (y/n)_**|  |
|**_Documentation Included (y/n)_**|  |

---
## Change Description

Example of implementation of the `-Wconversion` warning fixes.

Mostly consist of `static_cast` to change from `long (unsigned) int` to `(unsigned) int`.

This is mainly to illustrate the impact of the warning flag on the current code.

Needs to be merged after https://github.com/fprime-community/fpp/pull/400 has been merged.

## Rationale

Example of implementation of one of the warning flags mentioned here: https://github.com/nasa/fprime/discussions/2588

## Testing/Review Recommendations

N/A

## Future Work

N/A

### CODE REVIEW

**Comment by JohanBertrand** [comment]
@LeStarch is it possible to trigger the CI please?

**Comment by LeStarch** [comment]
@JohanBertrand let me fix the conflict, then sure!

**Comment by LeStarch** [comment]
@JohanBertrand, it has been triggered.  Thanks for the reminder!

**Comment by JohanBertrand** [comment]
@LeStarch I think most of the warnings are corrected for the Linux side, except the one related to fpp (https://github.com/fprime-community/fpp/pull/400). I'm not sure what is the path forward for that.

For the ones related to macOS, is there a good way for me to test it without waiting for the CI?

Also, just to make sure, are you interested into merging this PR in the future? It is a lot of (small) changes. Mostly about casting, but some small bugs are also fixed.

**Comment by LeStarch** [comment]
@JohanBertrand I do want this PR to be merged!  The above FPP PR is marked "Draft"....so you should probably make it a full PR and we can review.

**Comment by LeStarch** [comment]
As for the mac-side things are:

1. Let CI run it
2. Ask one of the maintainers to work through mac issues

**Comment by LeStarch** [comment]
@JohanBertrand this is on me because I didn't connect the dots between this PR and other work we have going on, but we have some fairly substantial type adjustments that we are working on.

In short NATIVE_INT_TYPE, NATIVE_UINT_TYPE are going to be replaced by correctly defined types for the use case.

As such, we expect collision with this work.  The thought on how to proceed would be:

1. Keep this PR open
2. Finish other work
3. Let one of the maintainers (e.g. myself) fix collisions and conflicts 
4. Merge this and the other work.

We are trying to not invalidate your work here, while also not leverage a bunch of extra work on you.

What are your thoughts on this?

**Comment by JohanBertrand** [comment]
I think only the fpp update is missing right now. I would have say that it's worth trying to review/merge everything now if it's ready, and try to merge it before the changes of types happen.

If you already started on it and you foresee some conflicts on your side, I'm happy following your approach and let this PR open until you're done with the types update.

File: Fw/Types/PolyType.cpp
Comment (JohanBertrand) [inline Fw/Types/PolyType.cpp]: I am not sure if that has any impact on other functions or components. Also, I did not find where this it currently used.

However, the unit tests are passing here because of the casts performed there. That's why it has not been seen before

**Comment by LeStarch** [comment]
> I think only the fpp update is missing right now. I would have say that it's worth trying to review/merge everything now if it's ready, and try to merge it before the changes of types happen.
> 
> If you already started on it and you foresee some conflicts on your side, I'm happy following your approach and let this PR open until you're done with the types update.

A fair point.

Our concern is masking.  You are, correctly, fixing errors, but these corrections might be better fixed with the new-methodology that was neither available nor known to you.  Your corrections may mask the other solution and make it harder for us to improve upon these issues.

However, we absolutely do not want to discard your work either.  These are valued contributions, and your counterpoint is essentially "don't let this code-rot", which is in itself abundantly valid.

Building on your recommendations I propose this:
1. We review your work and note anything that may be a mask, which we should revisit in the upcoming work.
2. Merge your work once the review has passed
3. Revisit point one as we perform our updates without blocking this PR

This solves both concerns:
1. Your work is merged without code-rotting
2. "Masking" issues are flagged, resolving our concerns.

*There may be many non-blocker comments from the review as we will be flagging issues for future consideration.*  We will clearly mark concerns that must be resolved before this merges, should anything come up!

Thanks for working with us.

**Comment by JohanBertrand** [comment]
I totally understand your point, and I am also concerned about masking issues through casting. 

I'm all good with your proposition. I am hoping that the merge of fpp will be solving the last warnings so that you can start the review.

Do you want me to add a temporary comment line before every casts I added to better identify it in the codebase later? That's something I can do easily, if that's helping you in the future.

Thank you very much!

**Comment by JohanBertrand** [comment]
The merge conflicts should be fixed, and the CI should pass in theory.

Please pay attention to the changes in the `Fw/Type` folder, since there was quite a few conflicts

File: Drv/LinuxUartDriver/LinuxUartDriver.cpp
Comment (LeStarch) [inline Drv/LinuxUartDriver/LinuxUartDriver.cpp]: For future review: m_allocationSize needs to be made into correct type.  Since it is a buffer allocation it should be U32.

File: Fw/Dp/DpContainer.cpp
Comment (LeStarch) [inline Fw/Dp/DpContainer.cpp:142]: For future review: should `setExtBuffer` take a `FwSizeType`

**Comment by LeStarch** [comment]
As my own note: here is the success criteria I am using for the review:

1. Cast to properly set type (e.g. FwAssertArgType): good, correct type chosen
2. Cast to API call specified type: good, obeying API
3. Cast to NATIVE_*_TYPE: good, will be caught in future review

Other items are flagged for future review.

**Comment by LeStarch** [comment]
@JohanBertrand unless you strongly object, I will go ahead and close out the macOS errors as I have access to the necessary OS to fix those.  Also, I have begun to review this.

File: Fw/Types/StringType.cpp
Comment (LeStarch) [inline Fw/Types/StringType.cpp]: For future review: do we need to cast to `size_t` just for a comparison to `0` below?  Perhaps we should just use the native result type from `strncmp`

File: Drv/Ip/IpSocket.cpp
Comment (LeStarch) [inline Drv/Ip/IpSocket.cpp:185]: For future review: why U32 here?  Seems arbitrary.

File: Drv/Ip/IpSocket.cpp
Comment (LeStarch) [inline Drv/Ip/IpSocket.cpp]: For future review: why U32 here?  Seems arbitrary.

File: Svc/CmdDispatcher/CommandDispatcherImpl.cpp
Comment (LeStarch) [inline Svc/CmdDispatcher/CommandDispatcherImpl.cpp:42]: For future review: should `slot` in the EVR match the iteration value of `U32`?

File: Utils/CRCChecker.cpp
Comment (LeStarch) [inline Utils/CRCChecker.cpp:198]: For future review: `PlatformIntType` usage here may need to be fixed.

File: Svc/StaticMemory/StaticMemoryComponentImpl.cpp
Comment (LeStarch) [inline Svc/StaticMemory/StaticMemoryComponentImpl.cpp]: I think this line needs a static cast to `FwAssertArgType`

File: Svc/Health/HealthComponentImpl.cpp
Comment (LeStarch) [inline Svc/Health/HealthComponentImpl.cpp:55]: For future review: this should iterate using `FwIndexType`

File: Svc/GenericHub/GenericHubComponentImpl.cpp
Comment (LeStarch) [inline Svc/GenericHub/GenericHubComponentImpl.cpp:96]: For future review: port should use `FwIndexType`

File: Svc/Deframer/Deframer.cpp
Comment (LeStarch) [inline Svc/Deframer/Deframer.cpp]: Another missing cast to assert arg type.

File: Svc/DpWriter/DpWriter.cpp
Comment (LeStarch) [inline Svc/DpWriter/DpWriter.cpp:198]: For future review: `U64` use seems suspicious.

File: Svc/ComLogger/ComLogger.cpp
Comment (LeStarch) [inline Svc/ComLogger/ComLogger.cpp]: Missing a cast.

File: Svc/ComQueue/ComQueue.cpp
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.cpp]: Cast

File: Svc/ComQueue/ComQueue.cpp
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.cpp]: For future review, why is entryIndex not `FwIndexType`?

File: Svc/ComQueue/ComQueue.cpp
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.cpp:169]: For future review: use of U32 in this iterator.  comQueueDepth.SIZE?

File: Svc/ComQueue/ComQueue.cpp
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.cpp:177]: For future review: use of U32 in this iterator. comQueueDepth.SIZE?

**Review by LeStarch — COMMENTED** [review]
Here are a list of comments for what I have reviewed thus far.  I see a few missing casts.  We also need to review `static_cast<FwIndexType>` as perhaps the variable type being cast should change.  

Nothing blocking except the need to fix CI.  I'll look at that tomorrow.

File: Svc/ComQueue/ComQueue.cpp
Comment (JohanBertrand) [reply Svc/ComQueue/ComQueue.cpp]: Just to make sure, do you want to have the cast  `static_cast<FwAssertArgType>()` even if it is a compatible/equivalent type?

**Comment by JohanBertrand** [comment]
@LeStarch Please go ahead with the fixes for macOS if you can, it takes too many iterations for me to solve all the warnings with the CI. 
Sorry for that.

Is there a problem with the CI stage `CI/Framework` ? I can't see the problem by looking at the CI log.

**Comment by LeStarch** [comment]
Yes there is.  This is a known issue with our older CI jobs (that predate Github actions).  Everything gets stuffed into a log file rather than the console.

If you click on "Summary" there is a zip archive one can download to look at the logs.  We need to fix this such that it is easier to see what the problem is.  Also, these artifacts may be restricted to maintainers...which could be another issue.

**Comment by JohanBertrand** [comment]
@LeStarch Thanks for the feedback. The issue should be fixed now, sorry for that.
I also fixed the error on the macOS CI at the same time.

**Comment by LeStarch** [comment]
Huzzah!

**Comment by JohanBertrand** [comment]
Awesome, thanks @LeStarch!

From the review, you want me to add the cast static_cast<FwAssertArgType>() even if it is a compatible/equivalent type, is that correct?

Also, do you want me to add a `// TODO: check type cast` in front of the comments you added for future review?

File: Svc/ActiveTextLogger/LogFile.cpp
Comment (LeStarch) [inline Svc/ActiveTextLogger/LogFile.cpp:73]: **For future review:** This is an established `U32` interface, but it likely should switch to `FwSizeType`?

File: Svc/AssertFatalAdapter/AssertFatalAdapterComponentImpl.cpp
Comment (LeStarch) [inline Svc/AssertFatalAdapter/AssertFatalAdapterComponentImpl.cpp:149]: **For future review:** these U32s need re-review.

File: Svc/BufferLogger/BufferLoggerFile.cpp
Comment (LeStarch) [inline Svc/BufferLogger/BufferLoggerFile.cpp:215]: **For future review:** U32 use in events.

File: Svc/Health/HealthComponentImpl.cpp
Comment (LeStarch) [inline Svc/Health/HealthComponentImpl.cpp:51]: **For future review:** is U32 tracking ping-entries appropriate?

File: Svc/ComQueue/ComQueue.cpp
Comment (LeStarch) [inline Svc/ComQueue/ComQueue.cpp:200]: **For future review:** is U32 as port number appropriate?

File: Svc/StaticMemory/StaticMemoryComponentImpl.cpp
Comment (JohanBertrand) [reply Svc/StaticMemory/StaticMemoryComponentImpl.cpp]: Fixed

File: Svc/Deframer/Deframer.cpp
Comment (JohanBertrand) [reply Svc/Deframer/Deframer.cpp]: Fixed

File: Svc/ComLogger/ComLogger.cpp
Comment (JohanBertrand) [reply Svc/ComLogger/ComLogger.cpp]: Fixed

File: Svc/ComQueue/ComQueue.cpp
Comment (JohanBertrand) [reply Svc/ComQueue/ComQueue.cpp]: Fixed

**Comment by JohanBertrand** [comment]
@LeStarch I solved the merge conflicts and added casts you requested in the review. Let me know if there is something else missing.

File: Svc/CmdSequencer/FPrimeSequence.cpp
Comment (LeStarch) [inline Svc/CmdSequencer/FPrimeSequence.cpp:191]: **For future review:** reconsider I32 as size in events.

File: Svc/ComQueue/ComQueue.cpp
Comment (LeStarch) [reply Svc/ComQueue/ComQueue.cpp:177]: Note: this is an autocoder problem, as array types use `U32` indices.

File: Svc/FileManager/FileManager.cpp
Comment (LeStarch) [inline Svc/FileManager/FileManager.cpp:183]: **For future review:** the use of `U32` here is probably incorrect.  Shell commands can return negative status.

File: Svc/PrmDb/PrmDbImpl.cpp
Comment (LeStarch) [inline Svc/PrmDb/PrmDbImpl.cpp:157]: **For future review:** types in events here.

File: Svc/DpCatalog/DpCatalog.cpp
Comment (LeStarch) [inline Svc/DpCatalog/DpCatalog.cpp:212]: **For future review:** why is `entry.dir` `FwIndexType` here and `FwSizeType` in the iterator.

**Review by LeStarch — APPROVED** [review]
I went ahead and fixed anything that seems like it **needed** to be fixed here.  Otherwise I left comments to review further.

Systematically, there seems to be no clear consensus in how to represent sizes in API calls and more specifically in model calls.  We should discuss a resolution and standards for this.

**Comment by LeStarch** [comment]
Note: I will fix CI this afternoon, as I broke it.

**Review by thomas-bc — APPROVED** [review]

**Pull request merged**
